tests/integration/pilot/routing_test.go

```diff
@@ -69,16 +75,18 @@ spec:
       request:
         add:
           istio-custom-header: user-defined-value`,
-			call: func() (echoclient.ParsedResponses, error) {
-				return apps.podA.Call(echo.CallOptions{Target: apps.podB, PortName: "http"})
-			},
-			validator: func(response echoclient.ParsedResponses) error {
-				return expectString(response[0].RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
+				call: func() (echoclient.ParsedResponses, error) {
+					return podA.Call(echo.CallOptions{Target: apps.podB[0], PortName: "http", Count: callCount})
+				},
+				validator: func(response echoclient.ParsedResponses) error {
+					return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
```

Comment:
nit: is there any cases we do not want to use response.Check? If not, should we just make the test code wrap the validator function?

Reply:
In traffic splitting, we have checks that require the entire ParsedResponses list.